### PR TITLE
Added select same tile tool.

### PR DESCRIPTION
**Pull request by IMMZ**
Added select same tile tool which selects all tiles which are similar to the current one. This pull request should cover [this feature request](https://github.com/bjorn/tiled/issues/973).

### CODE REVIEW

**Comment by kheftel** [comment]
excited for this :)

File: src/tiled/selectsametiletool.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    QRegion resultRegion;
+    if (tileLayer->contains(tilePos)) {
+        const Cell matchCell = tileLayer->cellAt(tilePos);
+
+        for (int i = 0; i < tileLayer->height(); ++i) {
+            for (int j = 0; j < tileLayer->width(); ++j) {
+                if (tileLayer->cellAt(i, j) == matchCell)
+                    resultRegion += QRegion(i, j, 1, 1);
+            }
+        }
```
Comment (bjorn) [inline src/tiled/selectsametiletool.cpp]: Building up a `QRegion` as a series of small 1x1 rectangles gets potentially quite slow. For this reason, a much more efficient version exists as `TileLayer::region(Condition condition)`. Please change your code to use that one. You can use the condition `MatchesAnyTile` in `tilesetdock.cpp` as an example.

File: src/tiled/selectsametiletool.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#include "tilelayer.h"
+
+namespace Tiled {
+namespace Internal {
+
+class MapDocument;
+
+/**
+ * Implements a tool that bucket fills (flood fills) a region with a repeatable
+ * stamp.
```
Comment (bjorn) [inline src/tiled/selectsametiletool.h]: Please update this comment. :-)

File: src/tiled/selectsametiletool.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/*
+ * selectsametiletool.cpp
+ * Copyright 2009-2010, Jeff Bland <jksb@member.fsf.org>
+ * Copyright 2010, Thorbjørn Lindeijer <thorbjorn@lindeijer.nl>
+ * Copyright 2010, Jared Adams <jaxad0127@gmail.com>
+ * Copyright 2011, Stefan Beller <stefanbeller@googlemail.com>
```
Comment (bjorn) [inline src/tiled/selectsametiletool.cpp]: You can give yourself some credit here. You can remove all the other names (also in the .h file).

**Comment by bjorn** [comment]
Please fix the AppVeyor build by also adding the new files to the `src/tiled/tiled.qbs` project file. This is also required for the Windows daily builds, which are done using Qbs.

File: src/tiled/selectsametiletool.cpp (hunk 1 above)
Comment (IMMZ) [inline src/tiled/selectsametiletool.cpp]: Fixed.

File: src/tiled/selectsametiletool.h (hunk 2 above)
Comment (IMMZ) [inline src/tiled/selectsametiletool.h]: Updated.

File: src/tiled/selectsametiletool.cpp (hunk 3 above)
Comment (IMMZ) [inline src/tiled/selectsametiletool.cpp]: Fixed.

File: src/tiled/selectsametiletool.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+#include "tilelayer.h"
+
+namespace Tiled {
+namespace Internal {
+
+class MapDocument;
+
+/**
+ * Implements a tool that bucket fills (flood fills) a region with similar tiles.
```
Comment (bjorn) [inline src/tiled/selectsametiletool.h]: This comment is still wrong, please read it carefully. This is not a fill tool. :-)

While you're at it, you may want to fix the documentation comment of the `MagicWandTool` as well...

Comment (IMMZ) [inline src/tiled/selectsametiletool.h]: My apologies. Fixed.

**Comment by bjorn** [comment]
I tested it shortly, seems to work fine. Nice one!

**Pull request merged**